### PR TITLE
load U2F js only on pages which need it

### PR DESCRIPTION
**Pull request by kdomanski**
Currently the U2F js will load on every page, as long as the user is enrolled in TOTP. WIth this change, the js will be only loaded on the U2F login screen and in the two-factor auth settings.

### CODE REVIEW

**Comment by kdomanski** [comment]
cc @silverwind

**Comment by lafriks** [comment]
Why not just set `RequireU2F` in needed handlers?

**Comment by CirnoT** [comment]
Yup, we should continue using `RequireU2F` but it should be removed from `loadSecurityData`.

It seems that it was added to `loadSecurityData` in order to populate `U2F` in `window.config` however I see no reason why it should be `true` on every page.

**Comment by silverwind** [comment]
My comment in https://github.com/go-gitea/gitea/pull/11573 was more aimed at the removal of the `RequireU2F` variable [here](https://github.com/go-gitea/gitea/pull/11573/files#diff-c9b5a1b845abb8d0bcebdaf23d497b61L29-L31) which would result in unconditional loading which should be avoided if possible.

I think the current loading mechanism via `RequireU2F` is generally fine (until we port the script to webpack/npm). The `window.config.U2F` variable is unused an can be dropped, it was added when I copied other similar variables to `window.config` in advance preparation.

**Comment by kdomanski** [comment]
Updated. Please restart the arm64 failing test.

**Comment by kdomanski** [comment]
ping @lafriks @silverwind

**Comment by silverwind** [comment]
Not sure I understand the impact of the move of this variable.

Is the`U2FRegistrations` variable not required in the new context?

**Comment by kdomanski** [comment]
The impact is that the U2F is set to load only on the security settings page and not wherever the security data is loaded, e.g. oauth endpoints.
It's inspired by your comment https://github.com/go-gitea/gitea/pull/11573#issuecomment-633030008

I'm pretty sure that the `U2FRegistrations` variable is orthogonal to this change.

**Comment by silverwind** [comment]
Maybe @6543 can test this? I don't have a security token, so I can't verify.

**Review by CirnoT — APPROVED** [review]

**Comment by CirnoT** [comment]
Works for me.

**Comment by 6543** [comment]
will test soon

**Comment by 6543** [comment]
ok cant test at the moment, it looks like my local test instance has no working U2F at all :/
tested with this branch and master - self cert https

I dont have time to dig into it at the moment - so neither a lgtm nor a block

**Comment by kdomanski** [comment]
@6543 What do you mean by "has no working U2F at all", what's going wrong? I'd love to help you test this.

**Comment by kdomanski** [comment]
@CirnoT If you could please suggest another reviewer, merging this PR would make it easier to deal with #11573

**Comment by CirnoT** [comment]
Not aware of anyone else aside from @6543, sorry.

**Comment by silverwind** [comment]
Maybe @jonasfranz can test.

**Comment by 6543** [comment]
I'll have some time today - Will hava a try again :)

**Comment by 6543** [comment]
hmm still get `Could not read your security key. ` with and without your patch :/

**Comment by kdomanski** [comment]
I've been getting a similar error when the base URL in the config was HTTP and not HTTPS.

File: templates/base/head.tmpl [hunk 1]
```diff
@@ -36,7 +36,8 @@
 			HighlightJS: {{if .RequireHighlightJS}}true{{else}}false{{end}},
 			SimpleMDE: {{if .RequireSimpleMDE}}true{{else}}false{{end}},
 			Tribute: {{if .RequireTribute}}true{{else}}false{{end}},
-			U2F: {{if .RequireU2F}}true{{else}}false{{end}},
+			Heatmap: {{if .EnableHeatmap}}true{{else}}false{{end}},
+			heatmapUser: {{if .HeatmapUser}}'{{.HeatmapUser}}'{{else}}null{{end}},
```
Comment (6543) [inline templates/base/head.tmpl]:
```suggestion
```

**Review by lafriks — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
make lg-tm work

**Pull request merged**